SU-HSDO/suhumsci
# Implement "Add Above" functionality in paragraphs

# READY FOR REVIEW

## Summary
- Install [Paragraphs Editor Enhancements](https://www.drupal.org/project/paragraphs_ee) and [Paragraphs Features](https://www.drupal.org/project/paragraphs_features)
- Replace Paragraph Browser with the default Paragraphs widget
- Enable the "Add Above" option
- Refactor Paragraph Browser custom code to work with the new widget
- Alter the paragraphs widget to remove the paragraph icons and change the add paragraph buttons to match previous version

## Need Review By (Date)
09/11

## Urgency
medium

## Steps to Test
For both Traditional and Colorful:

1. Create new Flexible and Private pages. Confirm that all component fields use the default paragraphs widget with the enhanced modal provided by `paragraphs_ee` (see screenshot below)
2. Add a Collection component to the Flexible page. Confirm that it also uses the new widget/modal
3. Edit an existing page with multiple components. Click the contextual menu in one of the paragraphs and select "Add Above". Confirm that the paragraphs selector modal is displayed and you can add a new paragraph in the expected position

<img width="1182" alt="Screenshot 2024-09-04 at 10 32 10 PM" src="https://github.com/user-attachments/assets/d35f6e8f-9039-452d-853f-4b4e08193a35">

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)
